tensorboard/webapp/header/plugin_selector_component.ng.html

```diff
@@ -17,14 +17,20 @@
 <mat-tab-group
   class="active-plugin-list"
   [selectedIndex]="getActivePluginIndex()"
-  (selectedTabChange)="onActivePluginSelectionChanged($event)"
   animationDuration="100ms"
 >
-  <mat-tab
-    *ngFor="let plugin of activePlugins"
-    [label]="plugin.tab_name"
-    [disabled]="!plugin.enabled"
-  >
+  <mat-tab *ngFor="let plugin of activePlugins" [disabled]="!plugin.enabled">
+    <ng-template mat-tab-label>
+      <!-- Manually subscribe to the click event on the tab content element.
+      Cannot trust the selectedTabChange event since it is async and can cause
+      infinte loop. -->
+      <span
+        class="plugin-name"
+        (click)="onActivePluginSelection($event, plugin.id)"
+      >
+        {{plugin.tab_name}}
```

Comment:
Naïve question, but from the way that this is structured it looks like
the click event would only be fired when the user clicks exactly on the
text rather than elsewhere on the tab; should the `(click)` listener be
on the `mat-tab` instead?

Reply:
What you are saying is correct but something really odd is going on. I think mat-tab-group is using non-conventional pattern and is kinda treating the element as a `ng-template`. Anyways, the click handler could not be bound correctly.

To work around the issue that you mentioned, I made sure the `span` take the width and height of the entire `mat-tab` element.

Reply:
Ah, I see. Okay. Well… if it works. :shipit: